docs/md/source.md

```diff
@@ -0,0 +1,186 @@
+updated rows and snapshots the rows to be processed downstream by an `Executor`.
+The table in your dataset must include a `DateTime` or `Timestamp` column
+representing the load or last modification date of that row to be compatible
+with a `Terra DataRepo` source. The `Terra DataRepo` source can only read inputs
+from a single table.
+
+When you `start` the workload, the `Terra DataRepo` source will start looking
+for new/updated rows from that instant.
+
+When you `stop` the workload, the `Terra DataRepo` source will stop looking
+for new/updated rows from that instant. All pending snapshots may continue
+to be processed by a later workload stage.
+
+!!! note
+    workflow-launcher creates snapshots of your data to be processed by a
+    later stage of the workload. Therefore, you must ensure the account
+    `workflow-launcher@firecloud.org` is a `custodian` of your dataset.
+    
+A typical `Terra DataRepo` source configuration in the workload request looks
+like:
+```json
+{
+  "name": "Terra DataRepo",
+  "dataset": "{dataset-id}",
+  "table": "{dataset-table-name}",
+  "column": "{dataset-column-name-to-poll}",
+  "snapshotReaders": [
+    "{user}@{domain}",
+    ...
+  ]
+}
+``` 
+The table below summarises the purpose of each attribute in the above request.
+
+| Attribute         | Description                                              |
+|-------------------|----------------------------------------------------------|
+| `name`            | Selects the `Terra DataRepo` source implementation.      |
+| `dataset`         | The `UUID` of dataset to monitor and read from.          |
+| `table`           | The name of the `dataset` table to monitor and read from.|
+| `column`          | The name of the UTC `DateTime` or `Timestamp` column in the `table` to poll.|
+| `snapshotReaders` | A list of email addresses whom should be `readers` of all snapshots created by workflow-launcher in this workload.|
+
+#### `dataset`
+The dataset attribute is the `UUID` that uniquely identifies the TDR dataset you
+want workflow-launcher to fetch workflow inputs form.
+
+#### `table`
+The `table` is the name of the table in the dataset that you want
+workflow-launcher to fetch inputs from. You should design this table such that
+each row contains all the inputs required to execute a workflow by the workload
+`Executor` downstream.
+
+#### `column`
+The `column` is the name of a column in the table specified above used to
+determine which rows are new and/or have been updated and therefore need
+reprocessing. It should be a `Timestamp`, but `DateTime` is accepted too.
+You must ensure that the `Timestamp` or `DateTime` column uses Universal 
+Coordinated Time (UTC).
+
+!!! note
+    Using a `Timestamp` will increase the likelihood of workflow-launcher
+    detecting and scheduling new rows in real-time due to greater precision.
+    Using `DateTime` may cause workflow-launcher to miss the row at first 
+    (though it will be picked up later).
```

Comment:
It wont - we should implement this!

Reply:
https://broadinstitute.atlassian.net/browse/GH-1397

Reply:
Agreed! We should defo implement this… Thanks for creating the ticket!